modules/cnf-configuring-fifo-priority-scheduling-for-ptp.adoc

```diff
@@ -0,0 +1,77 @@
+[id="cnf-configuring-fifo-priority-scheduling-for-ptp_{context}"]
+= Configuring FIFO priority scheduling for PTP hardware
+
+In telco or other deployment configurations that require low latency performance, PTP daemon threads run in a constrained CPU footprint alongside the rest of the infrastructure components. By default, PTP threads run with the `SCHED_OTHER` policy. Under high load, these threads might not get the scheduling latency they require for error-free operation.
+
+To mitigate against potential scheduling latency errors, you can configure the PTP Operator `linuxptp` services to allow threads to run with a `SCHED_FIFO` policy. If `SCHED_FIFO` is set for a `PtpConfig` CR, then `ptp4l` and `phc2sys` will run in the parent container under `chrt` with a priority set by the `ptpSchedulingPriority` field of the `PtpConfig` CR.
+
+[NOTE]
+====
+Setting `ptpSchedulingPolicy` is optional, and is only required if you are experiencing latency errors.
+====
+
+.Procedure
+
+. Edit the `PtpConfig` CR profile:
++
+[source,terminal]
+----
+$ oc edit PtpConfig -n openshift-ptp
+----
+
+. Change the `ptpSchedulingPolicy` and `ptpSchedulingPriority` fields:
++
+[source,yaml]
+----
+apiVersion: ptp.openshift.io/v1
+kind: PtpConfig
+metadata:
+  name: <ptp_config_name>
+  namespace: openshift-ptp
+...
+spec:
+  profile:
+  - name: "profile1"
+...
+    ptpSchedulingPolicy: SCHED_FIFO <1>
+    ptpSchedulingPriority: 65 <2>
+----
+<1> Scheduling policy for ptp4l and phc2sys processes. Use `SCHED_FIFO` on systems that support FIFO scheduling.
+<2> Required. Sets the integer value 1-65 used to configure FIFO priority for `ptp4l` and `phc2sys` processes.
+
+. Save and exit to apply the changes to the `PtpConfig` CR.
+
+.Verification
+
+. Get the name of the `linuxptp-daemon` pod and corresponding node where the `PtpConfig` CR has been applied:
++
+[source,terminal]
+----
+$ oc get pods -n openshift-ptp -o wide
+----
++
+.Example output
+[source,terminal]
+----
+NAME                            READY   STATUS    RESTARTS   AGE     IP            NODE
+linuxptp-daemon-gmv2n           3/3     Running   0          1d17h   10.1.196.24   compute-0.example.com
+linuxptp-daemon-lgm55           3/3     Running   0          1d17h   10.1.196.25   compute-1.example.com
+ptp-operator-3r4dcvf7f4-zndk7   1/1     Running   0          1d7h    10.129.0.61   control-plane-1.example.com
+----
+
+. Check that the `ptp4l` process is running with the updated `chrt` FIFO priority:
++
+[source,terminal]
+----
+$ oc -n openshift-ptp logs linuxptp-daemon-lgm55 -c linuxptp-daemon-container|grep chrt
+----
++
+.Example output
+[source,terminal]
+----
+...
+I1216 19:24:57.091872 1600715 daemon.go:285] /bin/chrt -f 65 /usr/sbin/ptp4l -f /var/run/ptp4l.0.config -2  --summary_interval -4 -m
+...
```

Comment:
Wondering if the ellipses above and below are really necessary? I suspect that more output is returned and that you want the user to only pay attention to this output snippet, so probably fine as you have it.